### PR TITLE
Switch build system to using poetry

### PR DESCRIPTION
**Pull request by bobmoretti**
Switch Python build system to use poetry instead of virtualenv/shell

I really hate a lot of things about virtualenv. Seems that the "poetry" project for Python is the least bad of all the alternatives.

This also (mostly) obviates the need for shell scripts. Unfortunately I had a hell of a time getting the github actions runner instances to run shell commands inside Python scripts, so I gave up on doing that and just duplicated some of the functionality inside of the action yaml.

I also somehow ended up updating a bunch of frontend stuff, which led me down a chain of fixing a bunch of eslint warnings and removing all explicit uses of `any` from the code. It's more type safe now.

Before we could close this PR I would need to update the developer docs.

### CODE REVIEW

**Comment by adamdoda** [comment]
We used to have something but I guess I've removed it.

File: doc/Development.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+## Introduction
+
+Tauntaun consists of two main parts: a backend implemented in Python and a frontend implemented in React/TypeScript. The backend uses the Quart server framework for network I/O. It uses PyDCS to load/save DCS .miz files as well as to provide abstractions for DCS mission file state. The frontend is implemented in React and uses (react-)esri-leaflet to provide mapping services.
+
+## TL;DR
+
+To get up and running, you need to install [Poetry](https://python-poetry.org/) and node/yarn. Then run the following commands
+
+```
```
Comment (adamdoda) [inline doc/Development.md:11]: Missing "git submodule update --init --recursive".

**Review by adamdoda — CHANGES_REQUESTED** [review]

File: doc/Development.md (hunk 1 above)
Comment (bobmoretti) [reply doc/Development.md:11]: Thanks

Comment (bobmoretti) [reply doc/Development.md:11]: Fixed via removing the submodule altogether 😛

**Comment by bobmoretti** [comment]
All tests passing, updated documentation, had to do something slightly awkward with the pydcs data files and pyinstaller, but I should be able to make a change upstream in pydcs to obviate that.

**Review by adamdoda — APPROVED** [review]

**Pull request merged**